openpype/modules/ftrack/plugins/publish/integrate_ftrack_instances.py

```diff
@@ -370,6 +377,124 @@ def json_obj_parser(obj):
+            thumbnail_data_items (list): List of thumbnail data items
+            are_multiple_synced_thumbnails (bool): If there are multiple synced
+                thumbnails
+
+        Returns:
+            dict: Thumbnail data item or empty dict
+        """
+        output_name = review_representation.get("outputName")
+        tags = review_representation.get("tags", [])
+        matching_thumbnail_item = {}
+        for thumb_item in thumbnail_data_items:
+            if (
+                are_multiple_synced_thumbnails
+                and (
+                    thumb_item["sync_key"] == output_name
+                    # intermediate files can have preset name in tags
+                    # this is usually aligned with `outputName` distributed
+                    # during thumbnail creation in `need_thumbnail` tagging
+                    # workflow
+                    or thumb_item["sync_key"] in tags
+                )
+            ):
+                # return only synchronized thumbnail if multiple
+                matching_thumbnail_item = thumb_item
+                break
+            elif not are_multiple_synced_thumbnails:
+                # return any first found thumbnail since we need thumbnail
+                # but dont care which one
+                matching_thumbnail_item = thumb_item
+                break
+
+        if not matching_thumbnail_item:
+            # WARNING: this can only happen if multiple thumbnails
+            # workflow is broken, since it found multiple matching outputName
+            # in representation but they do not align with any thumbnail item
+            self.log.warning(
+                "No matching thumbnail item found for output name "
+                "'{}'".format(output_name)
+            )
+            if not thumbnail_data_items:
+                self.log.warning(
+                    "No thumbnail data items found"
+                )
+                return {}
+            # as fallback return first thumbnail item
+            return thumbnail_data_items[0]
+
+
+        return matching_thumbnail_item
+
+    def _make_extended_component_name(self, component_item, repre, iteration_index):
+        """ Returns the extended component name
+
+        Name is based on the asset name and representation name.
+
+        Args:
+            component_item (dict): The component item dictionary.
+            repre (dict): The representation dictionary.
+            iteration_index (int): The index of the iteration.
+
+        Returns:
+            str: The extended component name.
+
+        """
+         # reset extended if no need for extended asset name
+        if self.keep_first_subset_name_for_review and iteration_index == 0:
+            return
+
+        # get asset name and define extended name variant
+        asset_name = component_item["asset_data"]["name"]
+        return "_".join(
+            (asset_name, repre["name"])
+        )
+
+    def _create_src_components(
```

Comment:
It creates single component, or?
```suggestion
    def _create_src_component(
```